src/vos/vos_common.c

```diff
@@ -457,7 +457,11 @@ vos_mod_init(void)
 	if (vos_start_epoch == DAOS_EPOCH_MAX)
 		vos_start_epoch = d_hlc_get();
 
-	rc = vos_pool_settings_init();
+	/**
+	 * Pass MD mode as int to enable custom umem allocator to be requested e.g. PMEM (0),
+	 * BMEM (1) and ADMEM (2). For MD on SSD specify (1) for now.
+	 */
+	rc = vos_pool_settings_init((int)bio_nvme_configured(SMD_DEV_TYPE_META));
```

Comment:
The vos_mod_init() is called before bio_nvme_init(), bio_nvme_configured() doesn't work at this moment. (it relies on nvme_conf parsing result in bio_nvme_init()).

So we need to move this vos_pool_settings_init() call after bio_nvme_init() for both engine mode and standalone mode.

Reply:
have fixed the order in stand-alone mode (vos_self_init()) looking at how to fix the engine mode case